prompter-source.lisp

```diff
@@ -790,18 +730,19 @@ If SIZE is NIL, capacity is infinite."
              (lambda (slot)
                (list (intern (symbol-name slot) "KEYWORD")
                      (slot-value object slot)))
-             (mopu:slot-names class-sym))))))
+             (slot-names class-sym))))))
 
 (defgeneric destroy (source)
   (:method ((source source))
-    ;; Ignore errors in case thread is already terminated.
-    ;; REVIEW: Is there a cleaner way to do this?
-    (ignore-errors (bt:destroy-thread (update-thread source)))
-    (ignore-errors (bt:destroy-thread (attribute-thread source))))
+    (maybe-funcall (destructor source) source)
+    (alex:when-let ((lpara:*kernel* (kernel source)))
+      (lpara:kill-tasks :default)
+      (lpara:end-kernel)
```

Comment:
Can you give an example?

Reply:
Roughtly this:
``` lisp
(defmethod (setf kernel) ((value null) (source source))
  (alex:when-let ((lpara:*kernel* (kernel source)))
    (lpara:kill-tasks :default)
    (lpara:end-kernel)
    (setf (slot-value source 'kernel) value)))
```

So that we don't have to kill tasks and end kernel manually every time.

Reply:
Problem is that the above appears as is only once + we should not always kill the tasks (because it's slow).  So it does not generalize very well.